docker-compose.gnu.yml

```diff
@@ -0,0 +1,77 @@
+version: '3.7'
+
+services:
+  postgres:
+    image: postgres:9.6
+    environment:
+      PGDATA: /var/lib/postgresql/data/pgdata
+      POSTGRES_USER: ${POSTGRES_USER}
+      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
+    ports:
+      - ${POSTGRES_PORT}:5432
+    restart: always
+    volumes:
+      - ./docker-scripts:/docker-entrypoint-initdb.d
+      - postgres_data:/var/lib/postgresql/data
+
+  graphql-engine:
+    image: hasura/graphql-engine:v1.0.0
+    ports:
+      - ${HASURA_PORT}:8080
+    restart: always
+    environment:
+      HASURA_GRAPHQL_ADMIN_SECRET: ${HASURA_SECRET}
+      HASURA_GRAPHQL_DATABASE_URL: postgres://${POSTGRES_USER}:${POSTGRES_PASSWORD}@postgres:${POSTGRES_PORT}/${HASURA_DATABASE}
+      HASURA_GRAPHQL_ENABLE_CONSOLE: ${HASURA_ENABLE_CONSOLE}
+      HASURA_GRAPHQL_ENABLED_LOG_TYPES: startup, http-log, webhook-log, websocket-log, query-log
+      HASURA_GRAPHQL_AUTH_HOOK: http://${DOCKER_HOST_IP}:8000/${HASURA_WEBHOOK}
```

Comment:
This seemed like a good initial solution, but it does create a bit of redundancy. I was going to suggest to keep it for the short-term, but it looks like an ideal solution may be easier. In the Makefile, if linux, we could set the docker_host_ip variable and, I think, pass it into docker with --env. See:

https://docs.docker.com/engine/reference/commandline/run/

Reply:
exporting env variable in the Makefile was my initial solution as well. however, I faced two issues that made me fall back to this one.
Issues with the first approach:
- Makefile shell commands are executed in a subshell, therefore, any export will not be propagated to parent shells instances and docker-compose cannot get docker_host_ip
- On Windows and Mac host.docker.intern ip is needed and shell script won't run on Windows, exporting docker_host_ip might not be portable